keps/sig-node/kubelet-node-endpoint.md

```diff
@@ -0,0 +1,133 @@
+
+For consistency the endpoint path "/node" looks similar to the "/pods" endpoint (no API versioning).
+
+## Other Solution
+
+A remediation solution has been already investigated and implemented but requires the deployment of an extra component.
+This component is working as a proxy/cache for queries to the `api-server` done by the daemonset pods.
+This solution resolves the initial problem of reducing pressure on the `api-server`, but in fact, it only moves the problem to a less sensitive component.
+
+This is the solution currently used by the Datadog monitoring solution deployment in Kubernetes via the `cluster-agent`.
+
+### Why /Node endpoint is a better solution
+
+Compared to the previous solution we think that using a `local` communication with the `kubelet` is a better solution and solves that scalability issue that we see with other presented solutions.
+
+With this solution, the load on the `api-server` from the daemonset pods does not depend on the number of nodes. And the load on each kubelet is very low.
+
+We have also seen the same approach, using a daemonset deployment that retrieved kubelet information, in other monitoring solutions.
+For example, GKE is deploying the `prometheus-to-sd` (stackdriver) daemonset pods that retrieve metrics on each kubelet separately.
+
+## Use Cases
+
+Node information, that is currently only available through the `api-server`, provides a lot of context for the metrics generated on application/process running in pods.
+
+### Pod metrics and logs tagging with Node labels or annotations
+
+Monitoring solutions based on the kubelet API ("/pods" and "/metrics") can decorate Pod metrics with additional context information; Information like the "Availability zone", "Provider", "Server type" etc are very useful for investigating issues.
+
+The current workaround is to retrieve that information from the API-Server, which generates a non-trivial amount of requests to the API-Server on a large Kubernetes cluster.
+
+### Container images monitoring on Node
+Knowing which Images are available on a specific node, and how much disk usage this represents can be useful for Node monitoring.
+However the current information present in the `Node.status.Images` can only be a subset of the images present on the node, and the images storage usage is not realistic since images can share layers, and as such use less storage than it is reported.
+
+Given the scope of the issue we describe here, proposing a solution for offering metrics on a node's images resource consumption should be done in a separate KEP.
+
+### Node Info and characteristics
+
+#### Container runtime
+
+Currently, knowing which container runtime is used is no easy task. You can retrieve it locally via the kubelet configuration file, but depending on the Kubernetes distribution, this file can be located in different places.
```

Comment:
Sig-node developed a tool, crictl, to facilitate querying the container runtime: https://github.com/kubernetes-sigs/cri-tools/blob/master/docs/crictl.md.  Alternatively, node agents could directly query the container runtime by doing something similar to what crictl does.

Reply:
Maybe check out `crictl info` to see if it has what you needs.